backstory/src/main/java/com/google/sps/story/PromptManagerBodyFactory.java

```diff
@@ -0,0 +1,228 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// s
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.sps.story;
+
+import com.google.sps.story.data.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Scanner;
+
+/**
+ * Constructs body of prompt given input keywords.
+ */
+public class PromptManagerBodyFactory {
+  /** Keywords for prompt generation */
+  private List<String> keywords;
+  /** Word processing client */
+  private PromptManagerWordTools wordTools;
+  /** randomness flag */
+  private boolean chooseRandomly;
+
+  /** Random number generator */
+  private Random templateChooser;
+
+  /** Data structures to hold classified words */
+  private List<String> gerunds;
+  private List<String> nouns;
+  private Map<WordType, List<String>> typeMap;
+
+  /** Minimum number of nouns for descriptive method. */
+  private final int MINIMUM_NOUNS = 3;
+
+  /** Final output string. */
+  private String outputBody;
+
+  /** Templates for introductory sentence completion */
+  private final String[] INTRO_TEMPLATES = {
+      "a <adj> <noun> as well as a <adj> <noun> decided to come together.",
+      "a <adj> <adj> <noun> and a <adj> <adj> <noun> appeared all at once. "};
```

Comment:
So, the array themselves are still mutable... would it be preferable to use an immutable list instead?
https://docs.oracle.com/javase/9/core/creating-immutable-lists-sets-and-maps.htm#GUID-202D195E-6E18-41F6-90C0-7423B2C9B381

Reply:
Done